tests/end-to-end/api/24-methods.js

```diff
@@ -189,4 +449,224 @@ describe('Meteor.methods', function() {
+					expect(res.body).to.have.property('status', 'error');
+					expect(res.body).to.have.property('message');
+				})
+				.end(done);
+		});
+
+		it('should return nothing when user doesnt have any permission', (done) => {
+			updatePermission('view-privileged-setting', []).then(() => {
+				updatePermission('edit-privileged-setting', []).then(() => {
+					updatePermission('manage-selected-settings', []).then(() => {
+						request.post(methodCall('private-settings:get'))
+							.set(credentials)
+							.send({
+								message: JSON.stringify({
+									method: 'private-settings/get',
+									params: [date],
+								}),
+							})
+							.expect('Content-Type', 'application/json')
+							.expect(200)
+							.expect((res) => {
+								expect(res.body).to.have.a.property('success', true);
+								expect(res.body).to.have.a.property('message').that.is.a('string');
+
+								const data = JSON.parse(res.body.message);
+								expect(data).to.have.a.property('result').that.is.an('array');
+								expect(data.result.length).to.be.equal(0);
+							})
+							.end(done);
+					});
+				});
+			});
+		});
+
+		it('should return properties when user has any related permissions', (done) => {
+			updatePermission('view-privileged-setting', ['admin']).then(() => {
+				updatePermission('edit-privileged-setting', []).then(() => {
+					updatePermission('manage-selected-settings', []).then(() => {
+						request.post(methodCall('private-settings:get'))
+							.set(credentials)
+							.send({
+								message: JSON.stringify({
+									method: 'private-settings/get',
+									params: [date],
+								}),
+							})
+							.expect('Content-Type', 'application/json')
+							.expect(200)
+							.expect((res) => {
+								expect(res.body).to.have.a.property('success', true);
+								expect(res.body).to.have.a.property('message').that.is.a('string');
+
+								const data = JSON.parse(res.body.message);
+								expect(data).to.have.a.property('result').that.is.an('object');
+								expect(data.result).to.have.a.property('update').that.is.an('array');
+								expect(data.result.update.length).to.not.equal(0);
+							})
+							.end(done);
+					});
+				});
+			});
+		});
+
+		it('should return properties when user has all related permissions', (done) => {
+			updatePermission('view-privileged-setting', ['admin']).then(() => {
+				updatePermission('edit-privileged-setting', ['admin']).then(() => {
+					updatePermission('manage-selected-settings', ['admin']).then(() => {
+						request.post(methodCall('private-settings:get'))
+							.set(credentials)
+							.send({
+								message: JSON.stringify({
+									method: 'private-settings/get',
+									params: [date],
+								}),
+							})
+							.expect('Content-Type', 'application/json')
+							.expect(200)
+							.expect((res) => {
+								expect(res.body).to.have.a.property('success', true);
+								expect(res.body).to.have.a.property('message').that.is.a('string');
```

Comment:
Same for this one

Reply:
If there are lines that you feel are too long (or get weirdly formatted in the diff view, consider making them smaller, so they are easier to read). If after the changes to the `then` you feel like they should be splitted, go ahead :D